src/main/java/network/brightspots/rcv/GuiTabulateController.java

```diff
@@ -208,13 +223,30 @@ private void watchParseCvrServiceProgress(Service<LoadedCvrData> service) {
           enableButtonsUpTo(tabulateButton);
           LoadedCvrData data = service.getValue();
           tabulateButton.setText("Tabulate " + String.format("%,d", data.numCvrs()) + " ballots");
+
+          // Populate the per-source CVR count table, and calculate the width of the filename column
+          perSourceCvrCountTable.getItems().clear();
+          int maxFilenameLength = 0;
+          for (ResultsWriter.CvrSourceData sourceData : data.getCvrSourcesData()) {
+            String countString = String.format("%,d", sourceData.getNumCvrs());
+            String fileString = new File(sourceData.source.getFilePath()).getName();
+            perSourceCvrCountTable.getItems().add(new Pair<>(fileString, countString));
+            maxFilenameLength = Math.max(maxFilenameLength, fileString.length());
+          }
+          perSourceCvrColumnFilepath.setPrefWidth(getEstWidthForStringInPixels(maxFilenameLength));
+          perSourceCvrCountTable.setVisible(true);
+
           data.discard();
           lastLoadedCvrData = data;
         };
 
     watchGenericService(service, onSucceededEvent);
   }
 
+  private float getEstWidthForStringInPixels(int stringLength) {
+    return stringLength * 7.5f;
```

Comment:
I wonder if this varies given OS, monitor specs, magnification accessibility settings, etc.? Is there no autosize option we could use for a column instead?

Reply:
There is a protected method that resizes, but doing that feels so much clunkier than this estimate: https://stackoverflow.com/a/38686402

This estimate is likely to make a cell that's too wide, but that's not really an issue. It should never make a cell that's too small. I think it's dependent on screen density but not any other factors, though I haven't tested on accessibility string scaling (and wonder if our UI works at all there?)

Note that the column is resizable so the estimate doesn't need to be perfect. LMK if you prefer the method in that stackoverflow link.

Reply:
Fine as-is, thanks!